models/login_source.go

```diff
@@ -731,8 +731,8 @@ func ExternalUserLogin(user *User, login, password string, source *LoginSource)
 	return user, nil
 }
 
-// UserSignIn validates user name and password.
-func UserSignIn(username, password string) (*User, error) {
+// UserSignIn validates user name and password. Password verification in LDAP skipped if already authenticated.
+func UserSignIn(username, password string, alreadyAuthenticated bool) (*User, error) {
```

Comment:
I guess one thing I don't really like about this is that the user has already signed in - we're just abusing this function to lookup the user in the external source - and only in one place (at least so far). Maybe we should be doing a ExternalUserLookup or something like that.

Reply:
Further if the user is in the db then this alreadyAuthenticated path fails.

Reply:
> Further if the user is in the db then this alreadyAuthenticated path fails.
This mod works ok for us in scenario with LDAP user backend + reverse proxy auth using HTTP header. No problems with initial account creation nor logging in/updating existing accounts from LDAP. If you see any bug please describe how to reproduce.

> Maybe we should be doing a ExternalUserLookup or something like that.
This mod was created for our needs with minimal changes in upstream code. Feel free to enhance it.